[rocprofiler-compute] Making torch trace an experimental feature (#3266)
ROCm/rocm-systems: projects/rocprofiler-compute/docs/how-to/profile/mode.rst

kernel executions with their originating PyTorch operators, providing insights into
which operators contribute to specific performance counter values.

> .. important::

Reviewer: This should be `.. warning::`

Author: Changed.

Requirements
------------

> * ``--experimental`` flag (torch-trace is an experimental feature)

Reviewer: Already mentioned before, lets remove this

* Valid PyTorch installation in the profiling environment
* PyTorch application must be run as a Python script or Python command

> * Workload's Python version must match the Python version ROCm installs roctx for

Reviewer 2: 'Workload's Python version must match roctx's Python version' would be clearer.

Author: Fixed.

Output
------

When Torch operator mapping is enabled, profiling generates additional output in

> the workload directory: ``marker_api_trace`` and ``counter_collection`` CSV files

Reviewer 2: the workload directory, namely ``marker_api_trace.csv`` and ``counter_collection.csv``.

are saved there (or under workload subdirectories depending on output format).
Correlation of operators with GPU kernels and consolidation into per-operator

> CSVs under ``torch_trace/`` is done after profiling (e.g. when running analyze).

Reviewer 3: ```suggestion
When Torch operator mapping is enabled, profiling generates additional CSV outputs (`marker_api_trace` and `counter_collection`) in the workload directory (or its subdirectories, depending on the output format).
Operator–kernel correlation is performed during post-processing (i.e., when running analyze), and the results are consolidated into per-operator CSV files under `torch_trace/`.
```

``torch_trace/`` directory

>    Per-operator CSV files. Each file is named from the **last component** of the

Reviewer 2: files. -> files:

Author: Removing implementation details.

>    Per-operator CSV files. Each file is named from the **last component** of the

Reviewer 3: ```suggestion
   Per-operator CSV files. Each file is derived from the **last component** of the
```

Author: Removing implementation details.

>    Per-operator CSV files. Each file is named from the **last component** of the

Reviewer: Such implementation details are subject to change and the user does not need to know all of this extra information. Please remove such details from public documentation.

Author: Removing implementation details.

   operator name (the part after the final ``/`` in hierarchical names), sanitized:

>    ``torch.`` removed, ``.`` replaced with ``_``. Examples: ``ones_like.csv``,

Reviewer 3: ```suggestion
   operator name (the portion after the final ``/`` in hierarchical names). The name is sanitized by removing the ``torch.`` prefix and replacing ``.`` with ``_``.
   Examples: ``ones_like.csv``,
```

Author: Removed the implementation details.

> .. note::

Reviewer: This should be under `.. warning::`

   * Torch trace is experimental: use ``rocprof-compute --experimental profile ...
     --torch-trace`` and ``rocprof-compute --experimental analyze ...`` with
     ``--list-torch-operators`` or ``--torch-operator`` as needed.

   * The ``--torch-trace`` option requires the application to be a Python command
     or Python script.

   * A valid PyTorch installation must be available in the environment where the
     workload runs.

   * The workload's Python version must match the Python version that ROCm installs

>      roctx for (see the roctx import error message if you see it).

Reviewer 2: Same comment as above.

Author: Fixed.
